gnd/src/main/java/com/google/android/gnd/ui/basemapselector/BasemapSelectorFragment.java

```diff
@@ -50,6 +66,63 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
   private void onMapReady(ExtentSelector map) {
     map.renderExtentSelectionLayer();
+
+    viewModel
+        .getTiles()
+        .observe(
+            this,
+            tiles -> {
+              map.updateExtentSelections(
+                  stream(tiles).map(Extent::fromTile).collect(toImmutableSet()));
+            });
+
+    viewModel
+        .getDownloadedExtents()
```

Comment:
Why separate the different extents by state? I'd think the ViewModel would just observe all the extends, and sync the states whenever any of them changes?

Reply:
Yep much simpler--haphazard midnight impl when scrambling for time. I've updated this according to your suggestion thanks!